Networking/Networking/Model/MetaData.swift

```diff
@@ -39,18 +47,92 @@ private extension MetaData {
     }
 }
 
-private extension AnyCodable {
-    func getAsString() -> String {
-        switch value {
-        case let string as String:
-            return string
-        default:
-            if JSONSerialization.isValidJSONObject(value) {
-                let data = (try? JSONSerialization.data(withJSONObject: value)) ?? Data()
-                return String(data: data, encoding: .utf8) ?? ""
-            } else {
-                return "\(value)"
+public struct MetaDataValue: Codable, Equatable, Sendable {
+    private let valueHolder: ValueHolder
+
+    public var stringValue: String {
+        valueHolder.stringValue
+    }
+    public var rawValue: String {
+        valueHolder.rawValue
+    }
+
+    public var isJson: Bool {
+        switch valueHolder {
+        case .string:
+            return false
+        case .json:
+            return true
+        }
+    }
+
+    public init(rawValue: String) {
+        let data = Data(rawValue.utf8)
+        if let jsonObject = try? JSONDecoder().decode(AnyCodable.self, from: data) {
+            valueHolder = try! ValueHolder(from: jsonObject.value)
+        } else {
+            valueHolder = .string(rawValue.removingPrefix("\"").removingSuffix("\""))
+        }
```

Comment:
Sorry to jump into the discussion. I would suggest using the following to avoid force unwrapping. 

```suggestion
        if let jsonObject = try? JSONDecoder().decode(AnyCodable.self, from: data),
           let value = try? ValueHolder(from: jsonObject.value) {
            valueHolder = value
        } else {
            valueHolder = .string(rawValue.removingPrefix("\"").removingSuffix("\""))
        }
```